<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name = "run_slam_optimization" default = "true"/>
<arg name = "run_slam_mapping" default = "true"/> 
<arg name = "run_dynamicobject_detection" default = "true"/> 
 <arg name="run_lidar_odometry"  default="true" />
  <arg name="run_getindnode"  default="true" />
  <arg name = "gps_convert" default = "false"/>  
  
  <arg name="configfile" default="$(find sensor_driver)/config/configmulti.xml" />
  <arg name="logdir" default="--log_dir=/home/zack-liu/catkin_yuanshi/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <param name="robot_description"
    textfile="$(find sensor_driver)/urdf/multivelodyne_3d.urdf" />

  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />


<group if="$(arg run_slam_mapping)">
  <node name="iv_slam_mapping_node" pkg="iv_slam_mapping_ros"
      type="iv_slam_mapping_node" args="
          -slam_configuration_directory $(find iv_slam_mapping_ros)/slam_configuration_files
          -slam_configuration_basename rciv_slam_fused.lua
--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)"          
      output="screen" >
    <remap from="points2" to="lidar_cloud_calibrated" />
    <!--remap from="points2_2" to="vertical_laser_3d" /-->
    <remap from="imu" to="imudata" />
  </node>

 <node pkg="ivrc_cross_communication" type="traversablearea_sub_main" name="traversablearea_sub_main"  output="screen"> 	
  </node>
</group>

<group if="$(arg run_slam_optimization)">
 <node name="iv_slam_optimization_node" pkg="iv_slam_optimization_ros"
      type="iv_slam_optimization_node" args="
          -slam_configuration_directory $(find iv_slam_optimization_ros)/slam_configuration_files
          -slam_configuration_basename rciv_slam_fused.lua
 --logtostderr=0  --colorlogtostderr=1 --stderrthreshold=0  $(arg logdir)"          
      output="screen">  
  </node>

<node name="rviz" pkg="rviz" type="rviz" required="true"
     args="-d $(find iv_slam_optimization_ros)/slam_configuration_files/iv_slam_optimization_ros.rviz" />

</group>
<group if="$(arg run_dynamicobject_detection)">
<node pkg="dynamicobject" type="dynamicobject" name="dynamicobject" args=" --logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2  $(arg logdir)" >
  <param name="heightdiff_threshold" value="0.3"/>
  </node>
</group>



  <node pkg="sensor_driver" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>


<group if="$(arg run_slam_mapping)">  
  <node pkg="covgrid_slam_ros" type="mappingmain" name="mappingmain" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" >
  <param name="configuration_directory" value="$(find covgrid_slam_ros)/config" />
  <param name="configuration_basename" value="localmap.lua" />
  <param name="withoutimu" value="true"/>
  <param name="withgps" value="true"/>
  <param name="sendflag" value="false"/>
  <param name="local_ip" value="192.168.0.112"/>
  <param name="local_port" value="9911"/>
  <param name="remote_ip" value="192.168.0.111"/>
  <param name="remote_port" value="9912"/>
  </node>

  <node pkg="covgrid_slam_ros" type="transformmain" name="transformmain" args="$(arg glogparam)" >
  <param name="heightdiff_threshold" value="0.3"/>
  </node>  
</group>
  
  <node pkg="sensor_driver" type="getmutivelodynedata" name="getmutivelodynedata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
  </node>
<group if="$(arg run_getindnode)">
  <node pkg="sensor_driver" type="getinsdata" name="getinsdata" args="$(arg glogparam)">
  </node>
</group>
<group if="$(arg gps_convert)">
 <node pkg="sensor_data_grabber" type="getgpsdata" name="getgpsdata">
  </node>

   
  </group>

</launch>
